### PR TITLE
[E129-MG < T1032-MG] Add MG_ENTERPRISE and license check

### PR DESCRIPTION
**Pull request by niko4299**

### CODE REVIEW

**Review by Josipmrden — APPROVED** [review]

**Comment by gitbuda** [comment]
The project compiles with `-DMG_ENTERPRISE=OFF`, is this ready to be open?

**Comment by niko4299** [comment]
> The project compiles with `-DMG_ENTERPRISE=OFF`, is this ready to be open?

File: src/auth/models.cpp [hunk 1]
```diff
@@ -407,43 +429,67 @@ const FineGrainedAccessPermissions &Role::GetFineGrainedAccessLabelPermissions()
...
     throw AuthException("Couldn't load role data!");
   }
   auto permissions = Permissions::Deserialize(data["permissions"]);
-  auto fine_grained_access_handler = FineGrainedAccessHandler::Deserialize(data["fine_grained_access_handler"]);
-  return {data["rolename"], permissions, std::move(fine_grained_access_handler)};
+#ifdef MG_ENTERPRISE
+  if (!data["fine_grained_access_handler"].is_object()) {
+    throw AuthException("Couldn't load user data!");
+  }
+  if (memgraph::utils::license::global_license_checker.IsValidLicenseFast()) {
```
Comment (gitbuda) [inline src/auth/models.cpp:460]: Check this

File: src/auth/models.cpp [hunk 2]
```diff
@@ -536,19 +600,33 @@ User User::Deserialize(const nlohmann::json &data) {
...
     throw AuthException("Couldn't load user data!");
   }
   auto permissions = Permissions::Deserialize(data["permissions"]);
-  auto fine_grained_access_handler = FineGrainedAccessHandler::Deserialize(data["fine_grained_access_handler"]);
-  return {data["username"], data["password_hash"], permissions, fine_grained_access_handler};
+#ifdef MG_ENTERPRISE
+  if (!data["fine_grained_access_handler"].is_object()) {
+    throw AuthException("Couldn't load user data!");
+  }
+  if (memgraph::utils::license::global_license_checker.IsValidLicenseFast()) {
```
Comment (gitbuda) [inline src/auth/models.cpp:611]: Check this

File: tests/unit/query_plan_create_set_remove_delete.cpp [hunk 3]
```diff
@@ -158,7 +162,10 @@ TEST(QueryPlan, CreateReturn) {
   EXPECT_EQ(1, CountIterable(dba.Vertices(memgraph::storage::View::OLD)));
 }
 
+#ifdef MG_ENTERPRISE
 TEST(QueryPlan, FineGrainedCreateReturn) {
+  memgraph::utils::license::global_license_checker.EnableTesting();
```
Comment (gitbuda) [inline tests/unit/query_plan_create_set_remove_delete.cpp:167]: Can this be moved to the Setup method?

Comment (gitbuda) [reply tests/unit/query_plan_create_set_remove_delete.cpp:167]: Aha, these tests are not using fixtures... do we add Setup?

**Review by BorisTasevski — APPROVED** [review]

**Pull request merged**